pyecharts/engine.py

```diff
@@ -152,17 +153,57 @@ def __init__(self, pyecharts_config=None, *args, **kwargs):
             *args,
             **kwargs)
 
-
-def render(template_file, notebook=False, **context):
+    def render_container_and_echarts_code(self, chart):
+        """
+        Render <div> and <script> code fragment for a chart.
+        :param chart:
+        :return:
+        """
+        tpl_string = """
+        {{ echarts_container(chart) }}
+        {{ echarts_js_content(chart) }}
+        """
+        tpl = self.from_string(tpl_string)
+        return tpl.render(chart=chart)
+
+    def render_chart_to_file(
+            self,
+            chart,
+            object_name='chart',
+            path='render.html',
+            template_name='simple_chart.html',
+            extra_context=None
+    ):
+        """
+        Render a chart or page to local html files.
+        :param chart: A Chart or Page object
+        :param object_name: Variable name for chart/page used in template
+        :param path: The destination file which the html code write to
+        :param template_name: The name of template file.
+        :param extra_context: A dictionary containing extra data.
+        :return:
+        """
+        context = {object_name: chart}
+        context.update(extra_context or {})
+        tpl = self.get_template(template_name)
+        html = tpl.render(**context)
+        utils.write_utf8_html_file(path, html)
+
+    def render_chart_to_notebook(self, charts, **context):
+        context.update({'charts': charts})
```

Comment:
charts could have been left inside context , hence this line can be avoided.

Reply:
For readability, make the *charts* a  required and positional parameter to match the name of this function.